Sources/SwiftDiagnostics/DiagnosticsFormatter.swift

```diff
@@ -0,0 +1,109 @@
+//===------------ DiagnosticsFormatter.swift - Formatter for diagnostics ----------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+
+public struct DiagnosticsFormatter {
+  
+  /// A wrapper struct for a source line and its diagnostics
+  private struct AnnotatedSourceLine {
+    var diagnostics = [Diagnostic]()
```

Comment:
Does this actually have to have a default value or could this be
```suggestion
    var diagnostics: [Diagnostic]
```

Reply:
I removed the default value. I added the struct because I think that we will add more information to a source line in the near future and maybe want to move part of the logic in there (e.g. `annotatedSourceLine.print()`).